controllers/admins.js

```diff
@@ -0,0 +1,40 @@
+const fs = require('fs');
+const admins = require('../data/admin.json');
+
+//Get all admins
+
+const getAdmins = (req, res) => {
+    res.status(200).json(
+        res.json(admins)
+    )
+}
+
+//Update an admin
+
+const updateAdmin = (req, res) => {
+    const found = admins.some(admin => admin.id === parseInt(req.params.id));
+
+    if (found){
+        admins.forEach(admin => {
+            if (admin.id === parseInt(req.params.id)){
+                !req.query.email ? null : admin.email = req.query.email;
+                !req.query.password ? null : admin.password = req.query.password
+            }   
+        })
+    }   else{
+        res.status(400).json({msg: `Admin with id: ${req.params.id} was not found`})
+    }
+
+    fs.writeFile('./data/admin.json', JSON.stringify(admins), {}, (error) => {
+        if (error){
+            res.status(400).send(error)
+        }   else{
+            res.status(200).json({msg: 'Admin updated', admins})
+        }
```

Comment:
just return the `res.status(..` to avoid use an `else` condition.

```
    fs.writeFile('./data/admin.json', JSON.stringify(admins), {}, (error) => {
        if (error){
            return res.status(400).send(error)
        } 
        return res.status(200).json({msg: 'Admin updated', admins})
    
    })
```

Reply:
Así esta bien ?


``` 
  fs.writeFile('./data/applications.json', JSON.stringify(Applications), {}, err => {
    if (err) {
      throw res.status(400).send(err)
    } 
    res.status(201).send(newApplication)
```
si ponia return en lugar de throw me salia un error. Lo que no se es si el throw funciona como el return, en el sentido que frena la funcion

Reply:
I get this error when I do that 
```
node:events:368
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::4000
    at Server.setupListenHandle [as _listen2] (node:net:1334:16)
    at listenInCluster (node:net:1382:12)
    at Server.listen (node:net:1469:7)
    at Function.listen (C:\Users\joaquin\Documents\Radium-Rocket\week-8\BaSD-MindSet\node_modules\express\lib\application.js:618:24)
    at Object.<anonymous> (C:\Users\joaquin\Documents\Radium-Rocket\week-8\BaSD-MindSet\index.js:17:5)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1361:8)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  code: 'EADDRINUSE',
  errno: -4091,
  syscall: 'listen',
  address: '::',
  port: 4000
}
[nodemon] app crashed - waiting for file changes before starting...
```
My code is 
```
const deletePosition = (req, res) => {
  const filteredPositions = Positions.filter(position => position.id !== req.query.id)
  const removedPosition = Positions.filter(position => position.id === req.query.id) 
  console.log(removedPosition)
  if(removedPosition.length === 0) res.status(404).send('Position not found')
  fs.writeFile('./data/positions.json', JSON.stringify(filteredPositions), {}, err => {
    if(err) {
      return res.status(400).send(err)
    } 
    return res.status(201).json(removedPosition)
  })
}
```

Reply:
Wow if I add a space before return it doesn't crash, and even if it's crashed it works on postman
```
const deletePosition = (req, res) => {
  const filteredPositions = Positions.filter(position => position.id !== req.query.id)
  const removedPosition = Positions.filter(position => position.id === req.query.id) 
  console.log(removedPosition)
  if(removedPosition.length === 0) res.status(404).send('Position not found')
  fs.writeFile('./data/positions.json', JSON.stringify(filteredPositions), {}, err => {
    if(err) {
      return res.status(400).send(err)
    } 
     return res.status(201).json(removedPosition) //the extra space is in this line
  })
}
```
now the server is running...

Reply:
I fixed it. I had another node terminal open. :)